core/src/main/java/org/apache/iceberg/LocationProviders.java

```diff
@@ -109,7 +109,7 @@ static class ObjectStoreLocationProvider implements LocationProvider {
 
     private static final HashFunction HASH_FUNC = Hashing.murmur3_32_fixed();
     private static final BaseEncoding BASE64_ENCODER = BaseEncoding.base64Url().omitPadding();
-    private final ThreadLocal<byte[]> temp = ThreadLocal.withInitial(() -> new byte[4]);
+    private transient ThreadLocal<byte[]> temp;
```

Comment:
I think we can have a `static` threadlocal and keep the initialization as is (and remove 175-177) to keep the change a bit simpler?
 Static fields (like transient) are also not serialized. Let me know if I'm missing something

Reply:
nice suggesgtion made the changes !